inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/service/DisplayMessageJobIntentService.kt

```diff
@@ -72,6 +82,23 @@ internal class DisplayMessageJobIntentService : JobIntentService() {
                                         message.getMessagePayload()?.resource?.imageUrl)))
     }
 
+    /**
+     * This method verifies campaign's contexts before displaying the message.
+     */
+    private fun verifyContexts(message: Message): Boolean {
+        val campaignContexts = message.getContexts()
+        if (!message.isTest() &&
+                campaignContexts.isNotEmpty() &&
+                verifyCampaignContextsCallback(campaignContexts, message.getMessagePayload()?.title ?: "")) {
+            // Message display aborted by the host app
+            Timber.tag(TAG).d("message display cancelled by the host app")
+            prepareNextMessage()
```

Comment:
Instead of calling `prepareNextMessage()` here, why not just add a loop inside `prepareNextMessage` and will go thru all the ready for displayed messages while `displayMessage` returns `false` (if not displayed).
What do you think?

Reply:
Sound better but what about `fetchImageThenDisplayMessage` method? It can't return any value synchronously